torch/csrc/jit/ir.h

```diff
@@ -394,6 +397,23 @@ struct Node : public Attributes<Node> {
     return node;
   }
 
+  // Add 'node' as an input to 'this' at the specified position in the
+  // arguments. Returns the added node for ease of chaining.
+  Value* insertInput(size_t i, Value* node) {
+    JIT_ASSERT(graph_ == node->owningGraph());
+    node->uses_.emplace_back(this, i);
+    inputs_.insert(inputs_.begin() + i, node);
+    for (size_t use_itr = i + 1; use_itr < inputs_.size(); ++use_itr) {
+      for (auto& use : inputs_[use_itr]->uses_) {
+        if (use.user == this) {
+          use.offset += 1;
+          break;
```

Comment:
On second thought, I think this might be incorrect. I think the conditional above needs to be `if (use == Use(this, use_itr)`. Otherwise in the case of multiple uses of the same value at different positions at the input of a node, we'll just be bumping the first instance first

Reply:
Right. I think you should use `findUseForInput` which is already implements this (just make sure to do it _before_ the call to insert).